panacea/query_client.go

```diff
@@ -291,20 +301,17 @@ func (q verifiedQueryClient) GetStoreData(ctx context.Context, storeKey string,
 	return result.Response.Value, nil
 }
 
-func (q verifiedQueryClient) getQueryBlockHeight(ctx context.Context) (int64, error) {
-	if height := ctx.Value(ContextKeyQueryBlockHeight{}); height != nil {
-		return height.(int64), nil
-	}
-	return q.GetLastBlockHeight(ctx)
+func (q *verifiedQueryClient) getQueryBlockHeight() int64 {
+	return q.cachedLastBlockHeight - 1
```

Comment:
Always do -1 in the last block, because the appHash is always in the next block. This is the same as the previous spec.